crates/p2p/src/swarm/message/serde.rs

```diff
@@ -25,13 +30,17 @@ pub(crate) mod pubkey_serializer {
 }
 
 pub(crate) mod signature_serializer {
-    use serde::{self, Deserializer, Serializer, de};
+    use serde::{self, Deserializer, Serializer, de, ser::SerializeSeq};
 
     pub(crate) fn serialize<S>(data: &[u8; 64], serializer: S) -> Result<S::Ok, S::Error>
     where
         S: Serializer,
     {
-        serializer.serialize_bytes(data)
+        let mut seq = serializer.serialize_seq(Some(64))?;
```

Comment:
(de-)serialization should be as copy less as possible.

1. we need to write custom (de-)serilization for bytes, not sequences
2. it should copy as less as possible. i.e. current deserialization Vec<u8> to Vec<u8> then gives to PublicKey::try_decode_protobuf(...) which gives a new instant of PublicKey which I guess is another copy.